### PR TITLE
feat(dapp): implement dapp create command

### PR DESCRIPTION
**Pull request by realdreamer**
https://app.asana.com/0/1205969628270714/1206000704279352

We have to publish create kadena app package. 

Working in cli found couple of minor issues in create kadena app package. I'll create a PR for that separately.

### CODE REVIEW

File: packages/tools/kadena-cli/src/dapp/commands/dappCreate.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  createCommand(
+    'create',
+    'create a new dapp project',
+    [globalOptions.dappTemplate()],
+    async (config, args) => {
+      const projectDir = join(process.cwd(), args[0]);
+      const { dappTemplate } = config;
+      const isFolderExists = existsSync(projectDir);
+
+      if (isFolderExists) {
```
Comment (jessevanmuijden) [inline packages/tools/kadena-cli/src/dapp/commands/dappCreate.ts:21]:
```
const folderExists = existsSync(projectDir);

      if (folderExists) {
```

File: packages/tools/kadena-cli/src/prompts/genericActionPrompts.ts [hunk 2]
```diff
@@ -35,3 +35,14 @@ async function actionHelper(message: string): Promise<string> {
     ],
   });
 }
+
+export async function actionAskForDappTemplate(): Promise<string> {
+  return await select({
+    message: 'What template do you want to use ?',
```
Comment (jessevanmuijden) [inline packages/tools/kadena-cli/src/prompts/genericActionPrompts.ts:41]: Maybe no space before question mark.

**Review by jessevanmuijden — APPROVED** [review]

**Comment by alber70g** [comment]
> [app.asana.com/0/1205969628270714/1206000704279352](https://app.asana.com/0/1205969628270714/1206000704279352)
> 
> We have to publish create kadena app package.
> 
> Working in cli found couple of minor issues in create kadena app package. I'll create a PR for that separately.

Can we add the create-kadena-app to the monorepo, BUT separate the tests for this project as an integration test, so that it doesn't block the CI (as this test is particularly slow)? @Ghislain89 should be able to assist you with that.

File: packages/tools/kadena-cli/src/utils/createCommand.ts [hunk 3]
```diff
@@ -180,7 +184,9 @@ export function createCommand<
           console.log('\n');
         }
 
-        await action(config, newArgs);
+        const generalArgs = rest.flatMap((r) => r.args);
+
+        await action(config, generalArgs, newArgs);
```
Comment (barthuijgen) [inline packages/tools/kadena-cli/src/utils/createCommand.ts:189]: I've discussed with @nillo. The third argument "newArgs" can just go away, as we don't use it currently.

File: packages/tools/kadena-cli/src/dapp/commands/dappCreate.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+export const createDappCommand: (program: Command, version: string) => void =
+  createCommand(
+    'create',
+    'create a new dapp project',
+    [globalOptions.dappTemplate()],
+    async (config, args) => {
+      const projectDir = join(process.cwd(), args[0]);
+      const { dappTemplate } = config;
+      const folderExists = existsSync(projectDir);
```
Comment (barthuijgen) [inline packages/tools/kadena-cli/src/dapp/commands/dappCreate.ts:19]: please use `services.filesystem.directoryExists` instead, we have this for testing purposes

**Review by nillo — APPROVED** [review]

**Review by nillo — APPROVED** [review]

**Pull request merged**